sonic-net/sonic-utilities
# Added support for tunnel route status in show vnet routes all.

#### What I did
This commit adds a new status filed in CLI "show vnet routes all"
#### How I did it
The status information is retrieved using the status of the prefix from Application DB.
The status field can have 3 values.
Empty : Configured but not in Application DB.
Active : Configured and active in ASIC.
Inacive : Configured but not in ASIC due to lack of BFD session with endpoint
#### How to verify it
Add a vxlan tunnel route with single Endpoint monitoring. run the command to see active/ inactive based on BFD session state.
if multiple Endpoints are present then atleast one has to be Up for status to show up as active.
if endpoint monitoring is not enabled then status would show up as active.
#### Previous command output (if the output of a command-line utility has changed)
vnet name        prefix                    endpoint                                     mac address    vni 
---------------  ------------------------  -------------------------------------------  -------------  -----
Vnet1            10.2.1.0/24               200.3.152.32                                                       
Vnet_v4_in_v4-0  150.62.191.1/32           100.251.99.1,100.251.99.2,100.251.99.3
Vnet_v6_in_v6-0  fddd:a150:a251::a6:1/128  fddd:a100:a251::a10:1,fddd:a101:a251::a10:1
test_v4_in_v4-0  160.62.191.1/32           100.251.7.1                                                        
test_v4_in_v4-0  160.63.191.1/32           100.251.7.2
test_v4_in_v4-0  160.64.191.1/32           100.251.7.3
#### New command output (if the output of a command-line utility has changed)
vnet name        prefix                    endpoint                                     mac address    vni    status
---------------  ------------------------  -------------------------------------------  -------------  -----  --------
Vnet1            10.2.1.0/24               200.3.152.32                                                       active
Vnet_v4_in_v4-0  150.62.191.1/32           100.251.99.1,100.251.99.2,100.251.99.3
Vnet_v6_in_v6-0  fddd:a150:a251::a6:1/128  fddd:a100:a251::a10:1,fddd:a101:a251::a10:1                        inactive
test_v4_in_v4-0  160.62.191.1/32           100.251.7.1                                                        inactive
test_v4_in_v4-0  160.63.191.1/32           100.251.7.2                                                        inactive
test_v4_in_v4-0  160.64.191.1/32           100.251.7.3                                                        inactive
